internal/fsutil/create_test.go

```diff
@@ -184,17 +194,23 @@ func (s *S) TestCreate(c *C) {
 		c.Assert(err, IsNil)
 		c.Assert(testutil.TreeDump(dir), DeepEquals, test.result)
 
-		if entry.Link != "" && entry.Mode&fs.ModeSymlink == 0 {
-			// Entry is a hardlink.
+		isHardLink := entry.Link != "" && entry.Mode&fs.ModeSymlink == 0
+		// Special case for hard links: use the same file info to compare the
+		// path and the link.
+		if isHardLink {
 			pathInfo, err := os.Lstat(entry.Path)
 			c.Assert(err, IsNil)
 			linkInfo, err := os.Lstat(entry.Link)
 			c.Assert(err, IsNil)
 			os.SameFile(pathInfo, linkInfo)
+		}
```

Comment:
Let's restore the `else` here per the other comment and keep the comment about the `fsutil.Create does not return ...` where it is applicable, so above the first `if`.

Reply:
Fixed in [4090b4c](https://github.com/letFunny/chisel/pull/17/commits/4090b4c4c27eda1bd386bb1757572a4529887ad9).